lib/index.js

```diff
@@ -262,9 +265,47 @@ module.exports = class DockerComposeParser {
   }
 
   /**
+   * Adds a main service if one doesn't already exist to the mains.builds object, and adds it to the services array
+   *
+   * @param {Service[]} services                  - Array of services
+   * @param {Object}    mains                     - Object containing all of the main builds and externals
+   * @param {Object}    mains.externals           - All services with external (github) builds (indexed by name)
+   * @param {Object}    mains.builds              - All services with local builds (same repo as compose file) (indexed by name)
+   * @param {Object}    opts
+   * @param {String}    opts.repositoryName       - Name of repository in GH or other service
+   * @param {String}    opts.userContentDomain    - User content domain for API environment. Used for creating hostnames.
+   * @param {String}    opts.ownerUsername        - Organization name in Runnable. Used for creating hostnames.
+   * @param {String}    opts.scmDomain            - Domain for used SCM. E.g. github.com
+   * @private
+   */
+  static addMainIfMissing (services, mains, opts) {
+    if (!Object.keys(mains.builds).length && !Object.keys(mains.externals.length)) {
+      const repositoryName = opts.repositoryName
+      const userContentDomain = opts.userContentDomain
+      const ownerUsername = opts.ownerUsername
+      const newMainService = {
+        metadata: {
+          name: repositoryName,
+          isMain: true,
+          envFiles: []
+        },
+        files: Parser.filesParser({ image: 'busybox' }),
```

Comment:
Can we add a specific tag to this, to make it a bit more predictable? This will prevent pulling of new `latest` images for an image we don't even use.